apps/web-portal-client/lib/features/auth/hooks/useAuth.jsx

```diff
@@ -1,28 +1,49 @@
 import {useContext, useCallback} from 'react'
 import {useDispatch, useSelector} from 'react-redux'
+import {useHistory} from 'react-router-dom'
+import {BrowserKeys, BrowserStorage} from '../../../utils'
 import {error, logging, login, logout} from '../slices'
 import {AppContext} from '../../../context'
 import {selectAuthError, selectAuthToken, selectLoginState} from '../selectors'
 
 export function useAuth() {
   const dispatch = useDispatch()
   const {api} = useContext(AppContext)
+  const history = useHistory()
+
+  const userLogin = (params) => {
+    dispatch(logging())
+    return api.auth
+      .login(params)
+      .then((result) => {
+        BrowserStorage.setLocalStorageValue(BrowserKeys.UserToken, result.token)
```

Comment:
as far as we already dispatch even with redux, better to not introduce yet one deps to browser storage

All side effects related browser storage is better to organize from 'other side', by middlware or just adding listeners for redux store (index.jsx)

Reply:
middleware usage will allow to move 'set/clear' token from the useAuth hook but 'read' token still will be in a checkSession  method